src/main/java/faang/school/postservice/service/PostServiceImpl.java

```diff
@@ -124,6 +159,47 @@ public List<PostDto> getAllPublishedPostsByProjectId(Long projectId) {
                 .map(postMapper::toDto)
                 .toList();
     }
+    
+    public List<ResourceDto> uploadImageToPost(Long postId, List<MultipartFile> files) {
+        log.info("Starting image upload for postId: {}, files count: {}", postId, files.size());
+        Post post = postRepository.findById(postId).orElseThrow(() -> new NotFoundException("Post doesn't exist"));
+        List<Resource> savedResources = new ArrayList<>();
+        try {
+            for (MultipartFile file : files) {
+                log.debug("Processing file: {} (size: {} bytes)", file.getOriginalFilename(), file.getSize());
+                Resource processedResource = proceedFile(file);
+                processedResource.setPost(post);
+
+                uploadToMinio(processedResource);
+                log.info("Successfully uploaded to MinIO: {}", processedResource.getKey());
+
+                Resource savedResource = resourceRepository.save(processedResource);
+                log.debug("Saved resource to DB with id: {}, key: {}",
+                        savedResource.getId(), savedResource.getKey());
+                savedResources.add(savedResource);
+            }
+        } catch (Exception e) {
+            log.error("Error during image upload. Rolling back changes", e);
+            savedResources.forEach(resource -> {
+                try {
+                    if (resource.getKey() != null) {
+                        minioClient.removeObject(
+                                RemoveObjectArgs.builder()
+                                        .bucket(bucketName)
+                                        .object(resource.getKey())
+                                        .build());
+                    }
+                } catch (Exception ex) {
+                    log.error("Failed to cleanup MinIO object: {}", resource.getKey(), ex);
+                }
+                resourceRepository.delete(resource);
+            });
+            throw new RuntimeException("Failed to upload images");
+        } finally {
+            savedResources.forEach(Resource::cleanup);
+        }
+        log.info("Successfully uploaded {} images for postId: {}", savedResources.size(), postId);
+        return savedResources.stream().map(resourceMapper::toDto).toList();
 
```

Comment:
```suggestion
}
```
после ребейза надо запускать проект)

Reply:
Согласен))) Пропустил шаг